tests/datasource_test.go

```diff
@@ -114,6 +138,77 @@ var _ = Describe("DataSource", func() {
 		_ = waitForReadyCondition(ds, corev1.ConditionFalse, "NoSource")
 	})
 
+	DescribeTable("[test_id:TODO] Labels should be copied to DataSource", func(sourceFn func() cdiv1.DataSourceSource, createSource func()) {
+		By("Create source for DataSource")
+		createSource()
+
+		By("Create DataSource")
+		ds := newDataSource(ds1Name)
+		ds.Spec.Source = sourceFn()
+		ds, err := f.CdiClient.CdiV1beta1().DataSources(ds.Namespace).Create(context.TODO(), ds, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		ds = waitForReadyCondition(ds, corev1.ConditionTrue, "Ready")
+
+		By("Check for labels on DataSource")
+		Eventually(func(g Gomega) {
+			ds, err := f.CdiClient.CdiV1beta1().DataSources(ds.Namespace).Get(context.TODO(), ds.Name, metav1.GetOptions{})
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(ds.Labels).To(HaveKeyWithValue(testKubevirtIoKey, testKubevirtIoValue))
+			g.Expect(ds.Labels).To(HaveKeyWithValue(testInstancetypeKubevirtIoKey, testInstancetypeKubevirtIoValue))
+			g.Expect(ds.Labels).To(HaveKeyWithValue(testKubevirtIoKeyExisting, testKubevirtIoNewValueExisting))
+		}, 60*time.Second, pollingInterval).Should(Succeed())
+	},
+		Entry("from DataVolume",
+			func() cdiv1.DataSourceSource {
+				return cdiv1.DataSourceSource{
+					PVC: &cdiv1.DataVolumeSourcePVC{Namespace: f.Namespace.Name, Name: pvc1Name},
+				}
+			},
+			func() {
+				createDv(pvc1Name, testURL(), map[string]string{
+					testKubevirtIoKey:             testKubevirtIoValue,
+					testInstancetypeKubevirtIoKey: testInstancetypeKubevirtIoValue,
+					testKubevirtIoKeyExisting:     testKubevirtIoNewValueExisting,
+				})
+			},
+		),
+		Entry("from PersistentVolumeClaim",
+			func() cdiv1.DataSourceSource {
+				return cdiv1.DataSourceSource{
+					PVC: &cdiv1.DataVolumeSourcePVC{Namespace: f.Namespace.Name, Name: pvc1Name},
+				}
+			},
+			func() {
+				createDv(pvc1Name, testURL(), map[string]string{
+					testKubevirtIoKey:             testKubevirtIoValue,
+					testInstancetypeKubevirtIoKey: testInstancetypeKubevirtIoValue,
+					testKubevirtIoKeyExisting:     testKubevirtIoNewValueExisting,
+				})
+				err := utils.WaitForDataVolumePhase(f, f.Namespace.Name, cdiv1.Succeeded, pvc1Name)
+				Expect(err).ToNot(HaveOccurred())
+				err = utils.DeleteDataVolumeOrphanPVC(f.CdiClient, f.Namespace.Name, pvc1Name)
```

Comment:
Seems a bit too clunky, maybe just create a populator standalone PVC? https://github.com/kubevirt/containerized-data-importer/blob/0f46feceb2fcefd1a8145c9773d92c6007b490f6/doc/cdi-populators.md?plain=1#L21

Reply:
ACK, will change it